src/frontends/lean/pp.cpp

```diff
@@ -447,6 +447,27 @@ bool pretty_fn::is_implicit(expr const & f) {
     }
 }
 
+bool pretty_fn::is_default_arg_app(expr const & e) {
+    if (m_implict || m_preterm)
+        return false; // showing default arguments
+    if (!closed(app_fn(e))) {
+        // the Lean type checker assumes expressions are closed.
+        return false;
+    }
+    try {
+        expr t = m_ctx.relaxed_whnf(m_ctx.infer(app_fn(e)));
+        if (is_pi(t)) {
+            expr arg_type = binding_domain(t);
+            t = binding_body(t);
+            if (!is_pi(t) && !is_var(t) && is_app_of(arg_type, get_opt_param_name(), 2)) {
+                expr defval = app_arg(arg_type);
+                return closed(defval) && defval == app_arg(e);
+            }
```

Comment:
We can use `m_ctx.is_def_eq(defval, app_arg(e))` instead of `defval == app_arg(e)`.